plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -859,19 +865,29 @@ public static Transaction newCreateTableTransaction(TrinoCatalog catalog, Connec
         SortOrder sortOrder = parseSortFields(schema, getSortOrder(tableMetadata.getProperties()));
 
         if (replace) {
-            return catalog.newCreateOrReplaceTableTransaction(session, schemaTableName, schema, partitionSpec, sortOrder, tableLocation, createTableProperties(tableMetadata, allowedExtraProperties));
+            return catalog.newCreateOrReplaceTableTransaction(
```

Comment:
nit: if you want to refactor the code, consider doing it in a preparatory commit.


why refactoring on line 868 and not in 877